custom_metrics/markup.js

```diff
@@ -0,0 +1,647 @@
+//[markup]
+// Uncomment the previous line for testing on webpagetest.org
+
+// Instructions for adding a new custom metric are in almanac.js.
+
+// output size ~ 1.5k to 2k
+
+var _logs = [];
+// saves the error details in the results log property.
+// returns the same error object so that it can be also used as the return value for a property.
+function logError(context, messageOrException, exception = null) {
+  let error = {type: "error", context: context};
+
+  try {
+    if (typeof messageOrException === 'string') {
+      error.message = messageOrException;
+    } 
+    else if (messageOrException instanceof Object) {
+      error.exception = messageOrException;
+
+      if (messageOrException.message) {
+        error.message = messageOrException.message;
+      }
+      else {
+        error.message = JSON.stringify(messageOrException, Object.getOwnPropertyNames(messageOrException));
+      }
+    }
+    else {
+      error.message = JSON.stringify(messageOrException);
+    }
+
+    if (exception) {
+      error.exception = exception;
+
+      if(exception.message) {
+        error.message += ": "+exception.message;
+      }
+    }
+  }
+  catch(e) {
+    error.message = "logError failed";
+    error.exception = e;
+  }
+
+  _logs.push(error);
+
+  return error;
+}
+
+var _custom_metrics = null;
+
+try { // whole process is placed in a try/catch so we can log uncaught errors
+
+  _custom_metrics = {
+    'doctype': document.doctype?.name ?? null,
+    'favicon': !!document.querySelector('link[rel*="icon"]'),
+    'viewport': document.querySelector('meta[name="viewport"]')?.getAttribute('content') ?? null,
```

Comment:
I think this is a duplicate of the existing [`meta_viewport.js`](https://github.com/HTTPArchive/legacy.httparchive.org/blob/master/custom_metrics/meta_viewport.js) custom metric. Could you check and remove this if so?

Reply:
removing